### PR TITLE
Update parsing from file so that we properly handle spaces and quotes…

### PR DESCRIPTION
**Pull request by tzakian**
## Description 

Updates how we parse into the string that we pass to Clap when working with a PTB file. Starts test organization, and added a couple initial tests.  

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui/tests/ptb_files/assign/assign_invalid.ptb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Raw assign with no value
```
Comment (stefan-mysten) [inline crates/sui/tests/ptb_files/assign/assign_invalid.ptb:1]: I keep forgetting we can add comments to ptb files :))

Comment (stefan-mysten) [reply crates/sui/tests/ptb_files/assign/assign_invalid.ptb:1]: It's obviously very useful to know what we're doing.

File: crates/sui/tests/ptb_files/assign/assign_valid.ptb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+--assign "a" "@0x0"
+# Can assign a variable to another variable
+--assign "b" "a"
+--make-move-vec "<address>" "[a,b]"
+# Can assign the result of the make-move-vec
+--assign "c"
+# Can assign to a field access
+--assign "c0" "c.0"
+# Can assign to the other field as well
+--assign "c1" "c.1"
```
Comment (stefan-mysten) [inline crates/sui/tests/ptb_files/assign/assign_valid.ptb:11]: These are very useful for the docs too. Are we complete here in the way we can use assign?

**Review by stefan-mysten — APPROVED** [review]
Excellent, love how the per folder organization of these tests.
Don't forget to remove the `:q` file before merging, it somehow just creeped in :D

File: crates/sui/src/ptb/ptb.rs [hunk 3]
```diff
@@ -365,6 +364,54 @@ impl PTB {
         builder.finish()
     }
 
+    // This function is used to split the input string into arguments. We decide when we have a new
+    // argument based on the different delimiters that we have. If we are inside of a delimiter and
+    // we hit a space, we should keep going until we find the end of the delimiter and a space at
+    // which point that is the end of the argument.
+    fn split_into_args(s: String) -> Vec<String> {
```
Comment (stefan-mysten) [inline crates/sui/src/ptb/ptb.rs:371]: While I know this is being tested through all the ptb test files, do you think it's a good idea to have a unit tests for this function as well?

File: crates/sui/tests/ptb_files/assign/assign_invalid.ptb (hunk 1 above)
Comment (tzakian) [reply crates/sui/tests/ptb_files/assign/assign_invalid.ptb:1]: Indeed it is :) Write (and comment) tests for future you is the way I view it

File: crates/sui/src/ptb/ptb.rs (hunk 3 above)
Comment (tzakian) [reply crates/sui/src/ptb/ptb.rs:375]: This is getting tested under a new `parsing` file under the tests, but yes we should make sure this is well tested, so if you spot any things that aren't tested feel free to shout

Comment (stefan-mysten) [reply crates/sui/src/ptb/ptb.rs:375]: Sounds good.

**Pull request merged**